trade-finance-manager-ui/server/helpers/map-currencies-to-radio-items.test.ts

```diff
@@ -0,0 +1,109 @@
+import { CURRENCY } from '@ukef/dtfs2-common';
+import { mapCurrenciesToRadioItems } from './map-currencies-to-radio-items';
+
+describe('map currencies to radio items helper', () => {
+  describe('mapCurrenciesToRadioItems', () => {
+    describe('when called without a checked currency', () => {
+      it('should return an array of RadioItem objects for all currencies', () => {
+        // Act
+        const currencyRadioItems = mapCurrenciesToRadioItems();
+
+        // Assert
+        expect(currencyRadioItems.length).toBe(Object.values(CURRENCY).length);
+        expect(currencyRadioItems).toEqual([
+          {
+            text: CURRENCY.GBP,
+            value: CURRENCY.GBP,
+            checked: false,
+            attributes: {
+              'data-cy': 'currency-GBP',
+            },
+          },
+          {
+            text: CURRENCY.EUR,
+            value: CURRENCY.EUR,
+            checked: false,
+            attributes: {
+              'data-cy': 'currency-EUR',
+            },
+          },
+          {
+            text: CURRENCY.USD,
+            value: CURRENCY.USD,
+            checked: false,
+            attributes: {
+              'data-cy': 'currency-USD',
+            },
+          },
+          {
+            text: CURRENCY.JPY,
+            value: CURRENCY.JPY,
+            checked: false,
+            attributes: {
+              'data-cy': 'currency-JPY',
+            },
+          },
+        ]);
+      });
+    });
+
+    describe('when called with a checked currency', () => {
+      it('should return an array of RadioItem objects with only the specified currency checked', () => {
+        // Arrange
+        const checkedCurrency = CURRENCY.GBP;
+
+        // Act
+        const currencyRadioItems = mapCurrenciesToRadioItems(checkedCurrency);
+
+        // Assert
+        expect(currencyRadioItems).toEqual([
+          {
+            text: CURRENCY.GBP,
+            value: CURRENCY.GBP,
+            checked: true,
+            attributes: {
+              'data-cy': 'currency-GBP',
+            },
+          },
+          {
+            text: CURRENCY.EUR,
+            value: CURRENCY.EUR,
+            checked: false,
+            attributes: {
+              'data-cy': 'currency-EUR',
+            },
+          },
+          {
+            text: CURRENCY.USD,
+            value: CURRENCY.USD,
+            checked: false,
+            attributes: {
+              'data-cy': 'currency-USD',
+            },
+          },
+          {
+            text: CURRENCY.JPY,
+            value: CURRENCY.JPY,
+            checked: false,
+            attributes: {
+              'data-cy': 'currency-JPY',
+            },
+          },
+        ]);
+      });
+    });
```

Comment:
Could we create some kind of "generic" or "base" expected currencies, will make these tests simpler e.g something like this:

```ts
const expected = [
  ...genericCurrencies,
  {
    text: CURRENCY.JPY,
    // etc,
    checked: true,
  },
];

expect(currencyRadioItems).toEqual(expected);

```

Reply:
Updated to use a new test helper thanks `aCurrencyRadioItem` (similar pattern to other test helpers)